miio/gateway.py

```diff
@@ -647,11 +689,23 @@ def device_type(self):
         """Return the device type name."""
         return self.type.name
 
+    @property
+    def name(self):
+        """Return the name of the device."""
+        if self._name == "unknown":
+            self._name = self.device_type
```

Comment:
I don't think it's worth to have this check here in place. When someone notices that there is an "Unknown (some sid)", I hope they'll report back so that can be added to the library.

Reply:
I have removed that check